galley/pkg/kube/source/init.go

```diff
@@ -15,18 +15,86 @@
 package source
 
 import (
+	"io/ioutil"
 	"fmt"
+	"os"
 	"time"
+	"strings"
 
+	"github.com/ghodss/yaml"
 	"github.com/hashicorp/go-multierror"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/wait"
-
 	"istio.io/istio/galley/pkg/kube"
 	kube_meta "istio.io/istio/galley/pkg/metadata/kube"
 )
 
+func writeCRDs(cs clientset.Interface, fileName string) (error) {
+	CRDData, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		scope.Infof("Read CRD %v", err)
```

Comment:
Here and below... 

Please use longer, more descriptor log comments (i.e. error reading CRD file: %v")
Also, please use appropriate log level methods (i.e. Infof => Errorf)

Reply:
sure - will have a more thourough review of the code myself.  I am really looking for an answer to "Is this the right place for this work".  It seems from your review your supportive of this implementation within galley.  The tradeoff here is that galley will, for at least 1.1, be a mandatory component (to instantiate CRDs).  We can make it optional again in 1.2 with this approach.

Reply:
I am generally supportive, yes (see a bit below for more in another comment).

What is the criteria for mandatory v.s. optional? With maturing & eventual switch to MCP model & Galley becoming stable, I think Galley will become a mandatory component for a typical K8s deployment. This is somewhat orthogonal to this particular discussion, but I want to make sure we're all on the same page, while we're making future-looking plans.